src/views/PokeDetails.js

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import * as pokemonServices from '../services/pokemons.service';
+import PokeHeader from '../sltyledComponents/pokemonHeader';
+import { selectPokemon } from '../actions/pokemonActions';
+import PokeTabs from '../sltyledComponents/pokeTabs';
+import styled from 'styled-components';
+import { colorsByType } from '../sltyledComponents/colors';
+
+
+const Description = styled.div`
+    width: 80%;
+    margin: 40px auto;
+    background: white;
+    border: 2px solid lightgray;
+    border-radius: 5px;
+    padding-bottom: 10px;
+`;
+
+function PokeDetails(props) {
+
+    const { id } = useParams();
```

Comment:
Taking the ID from params is the correct way. 
But you are also trying to get it from redux (pokemon.id?) which leads to confusion.